src/libs/actions/Session/index.js

```diff
@@ -739,6 +749,28 @@ function validateTwoFactorAuth(twoFactorAuthCode) {
     API.write('TwoFactorAuth_Validate', {twoFactorAuthCode}, {optimisticData, successData, failureData});
 }
 
+/**
+ * Waits for a user to sign in.
+ *
+ * If the user is already signed in (`authToken` is truthy), the promise resolves immediately.
+ * Otherwise, the promise will resolve when the `authToken` in `ONYXKEYS.SESSION` becomes truthy via the Onyx callback.
+ *
+ * @returns {Promise<boolean>} A promise that resolves to `true` once the user is signed in.
+ * @example
+ * waitForUserSignIn().then(() => {
+ *   console.log('User is signed in!');
+ * });
+ */
+function waitForUserSingIn() {
+    return new Promise((resolve) => {
```

Comment:
Shouldn't we add a `reject` case? what happens if the sign in fails?

Reply:
it can't fail! :) You are either logged in and it will resolve immediately, or you are not logged in and it will resolve after you'll log in. There is no other scenario. This function doesn't cover a scenario of a failed login attempt, because it's not designed for that.